frontend/src/scenes/session-recordings/player/list/consoleLogsUtils.tsx

```diff
@@ -0,0 +1,112 @@
+import React from 'react'
+import { findAllIndices } from 'lib/utils'
+
+const STRING_INCLUDES_URL = new RegExp(
+    '([a-zA-Z0-9]+://)?([a-zA-Z0-9_]+:[a-zA-Z0-9_]+@)?([a-zA-Z0-9.-]+\\.[A-Za-z]{2,4})(:[0-9]+)?(/.*)?'
+)
+
+export interface ParsedEntry {
+    type: 'array' | 'object' | 'string'
+    parsed: Array<any> | Record<string, any> | React.ReactNode
+    size: number
+    traceUrl?: React.ReactNode // first url in stack
+}
+
+// Parses single payload entry
+//  - If url is detected in the string, shorten it and wrap with <a> tag
+//  - If object is detected in string, parse it and make it pretty
+//  - If array is detected in string, parse it and make it pretty
+export function parseEntry(entry: string): ParsedEntry {
+    // If entry is flanked by `"`'s, remove them.
+    let rawEntry = entry.replace(/^"/, '').replace(/"$/, '')
+    let traceUrl: React.ReactNode = null
+
+    // Check if object or array
+    try {
+        const parsedObject = JSON.parse(rawEntry)
+        const isArray = Array.isArray(parsedObject)
+        return {
+            type: isArray ? 'array' : 'object',
+            parsed: parsedObject,
+            size: isArray ? parsedObject.length : Object.keys(parsedObject).length,
+        }
+    } catch {
+        /* Silently catch errors and continue to parse entry as string */
+    }
+
+    // Align all whitespace
+    rawEntry = rawEntry.replace(/\\n/g, ' ').replace(/\s+/g, ' ')
+
+    // Wrap urls with anchor tags
+    const boluses = rawEntry.split(' ').map((bolus) => {
```

Comment:
What does boluses mean?

Reply:
"a small rounded mass of a substance, especially of chewed food at the moment of swallowing" but i can see how thats not appropriate in software engineering lol